ui/src/components/mobileservices/validators/AbstractBindingValidator.js

```diff
@@ -0,0 +1,68 @@
+/* eslint no-continue: off */
+/* eslint class-methods-use-this: off */
+export class AbstractBindingValidator {
+  constructor(platform, customProperties = {}) {
+    this.customProperties = customProperties;
+
+    this.detectPlatform = customProperties.platformDetector;
+    this.errorsKey = customProperties.errorField;
+    this.platform = platform;
+  }
+
+  addError(errors, key, message) {
+    errors[this.errorsKey || key].addError(message);
+  }
+
+  preFieldValidation(key, value) {
```

Comment:
this one isn't used anywhere. Do we need it? When the `preFieldValidation` should be used? What's the difference between the preValidate/validateField/postValidate fields?

Reply:
The execution order is:

1. preFieldValidation
2. validateField
3. postFieldValidation

both `preFieldValidation` and `postFieldValidation` are optional. The use of them is to perform some operation *before* the validation (`preFieldValidation`) or *after* the validation. If any of the method fails, the flow is interrupted.

Currently, only the `postFieldValidation` is used: if the `validateField` for the password field passes, then the password is matched to the `password confirmation`.

Since the confirmation field is a field that we add in the UI, it is not part of the `errors` object and custom code must be executed to get its value and validate it: the `postFieldValidation` helps in this case (see [here](https://github.com/aerogear/mobile-developer-console/blob/caabe8e32372a6166a770e04575aaf6e58a92d5d/ui/src/components/mobileservices/BindingPanel.js#L144))

Reply:
I can understand, but personally I think it is a bit over-complicate: effectively they can all processed as part of validate, so I just don't get the benefit of splitting them. 

Anyway, given that `preFieldValidation` is not used anywhere, it's best to remove it for now. We can add it back later if we really need it.

Can we also add some comments in here to explain the difference.